nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExecuteProcess.java

```diff
@@ -181,7 +183,14 @@ public Thread newThread(final Runnable r) {
 
     @OnUnscheduled
     public void shutdownExecutor() {
-        executor.shutdown();
+        try {
+            executor.shutdown();
+        } finally {
+            if (this.externalProcess.isAlive()) {
+                this.getLogger().warn("Interrupting external process since it did not finished at the time stop Processor was initiated.");
```

Comment:
_Process hasn't terminated, forcing the interrupt._ - that is what I am putting in now since we are actually issuing a request to stop something that has not terminated by itself yet. So, I've removed "on request".